Automattic/pocket-casts-android
# Support edge-to-edge on the podcasts list page

## Description

There will be multiple pull requests required to support Android 15 edge-to-edge but I'm going to break them down into smaller PRs so they are easier to review and any mistakes and feedback happen early. I will merge them all into a `task/android-15` branch and once complete raise a PR to merge that into `main`.

This change focuses on the following:
- Making the changes so the app runs
- Focusing on the podcasts list page

Please ignore other areas of the app for now. 

## Testing Instructions
1. Subscribe to enough podcasts that the podcasts list can scroll (turn off auto download first)
2. Open the podcasts tab
3. ✅ Verify the status bar matches the toolbar theme color and that the page isn't overlapping it.
4. Long press on the toolbar to change the theme
5. ✅ Verify the different themes look good
6. ✅ Verify you can see the end of the list
7. Play an episode so the mini player is shown
8. ✅ Verify you can still see the end of the list
9. Change the navigation bar height by going to system settings -> Display & touch -> Navigation mode -> 3-button navigation
10. ✅ Verify the end of the list again
11. Rotate into landscape
12. ✅ Verify the content isn't hidden under and controls such as a camera

Try this on some different version devices from SDK 24 to 35.

## Screenshots

Android 15, SDK 35

| Before | After |
| --- | --- |
| ![old-podcasts-full](https://github.com/user-attachments/assets/64c81a3e-5529-42f0-949c-2c4f1d6db2f9) | ![position1](https://github.com/user-attachments/assets/88d49aa0-21aa-4f81-a5c4-f27c91e19704) |
| ![old-podcasts-snack-bar](https://github.com/user-attachments/assets/8353b177-7ae4-485f-8ed4-e4e277383ddf) | ![position2](https://github.com/user-attachments/assets/49a54e7e-88c4-4002-8c2d-dea00ef2bff3) |
| ![old-podcasts-list-bottom](https://github.com/user-attachments/assets/9c37ec0d-8564-477f-ba9f-684623c6fda6) | ![position3](https://github.com/user-attachments/assets/f4246061-91bf-40d0-b920-302739f2cdf4) |
| ![old-podcasts-dark](https://github.com/user-attachments/assets/fdadbf20-bc5b-46f0-8ef4-df19068f183e) | ![position4](https://github.com/user-attachments/assets/3495661d-6f18-4b67-9cc9-622791f0a11e) |
| ![old-podcasts-pink](https://github.com/user-attachments/assets/1eb6283a-42d6-49ce-9ced-b48331137ead) | ![position5](https://github.com/user-attachments/assets/ec6427c0-fc0f-49a7-9cf8-24cc447a67e2) |
| ![old-podcasts-indigo](https://github.com/user-attachments/assets/c7bfca9e-cc9d-4364-a5a6-b437f641afd6) | ![position6](https://github.com/user-attachments/assets/05367fe4-360c-45e8-b77a-0806c15700bd) |
| ![old-podcasts-extra-dark](https://github.com/user-attachments/assets/13350c1b-d547-41cf-946d-f7620831c037) | ![position7](https://github.com/user-attachments/assets/38cded5a-5e0c-4227-864c-3628e39fec63) |
| ![old-podcasts-high-contrast](https://github.com/user-attachments/assets/a6516070-60bf-44a9-90f7-d955883aa594) | ![position8](https://github.com/user-attachments/assets/0e5c1788-27e9-4e0f-915a-90ad50644a9b) |
| ![old-podcasts-electric](https://github.com/user-attachments/assets/b62340e9-d27f-4dc6-83fb-7ebbfd9dcb8e) | ![position9](https://github.com/user-attachments/assets/7fcf63dc-15b1-4b30-bb8e-c1c8649b4d59) |
| ![old-podcasts-classic](https://github.com/user-attachments/assets/5c4f9225-714c-459e-9b1d-94dd936f1d17) | ![position10](https://github.com/user-attachments/assets/a2dffe28-db48-4a71-a2ca-cdd6afdc4c18) |
| ![old-podcasts-radioactive](https://github.com/user-attachments/assets/05143da3-aa21-4e1a-9f7c-a7546bcf64be) | ![position11](https://github.com/user-attachments/assets/a377c72c-8689-4101-9382-0874200fe09b) |
| ![old-podcasts-landscape](https://github.com/user-attachments/assets/ff8651ad-5d7d-449c-b65f-c0333381674f) | ![position1](https://github.com/user-attachments/assets/b7bc17a5-1b02-4bc0-8f91-a1dff60fea96) |
| ![old-podcasts-empty](https://github.com/user-attachments/assets/e1a420d0-d8e8-4b38-af5b-76336b8077c5) | ![Screenshot_20241118_080458](https://github.com/user-attachments/assets/722c4615-e65d-404b-86f8-52b9fcf75d4f) |

Android 14, SDK 34

| Before | After |
| --- | --- |
| ![before1](https://github.com/user-attachments/assets/77260478-9214-4e4c-9f94-d1f32c0f84a3) | ![after1](https://github.com/user-attachments/assets/36f955e8-4f84-445a-9c65-b74099947643) |
| ![before2](https://github.com/user-attachments/assets/764c300d-9226-4bfd-997e-579ee2b4d184) | ![after2](https://github.com/user-attachments/assets/a14ba605-852e-4a49-80b8-868dd1736729) |
| ![before3](https://github.com/user-attachments/assets/b35b2724-eb45-423f-819e-97bb37672e58) | ![after3](https://github.com/user-attachments/assets/41c818a0-08b6-4cc0-a1c8-8b622e775c78) |

Android 8, SDK 26

| Before | After |
| --- | --- |
| ![before-android-26-light](https://github.com/user-attachments/assets/5ec77b05-3ca0-4712-8638-c241b5b761e2) | ![after-android-26-light](https://github.com/user-attachments/assets/fd2a2a95-b9ca-4467-9ba1-9a5cae190461) |
| ![before-android-26-dark](https://github.com/user-attachments/assets/253b2c65-5b95-44fa-bf2d-d2ad37b63a96) | ![after-android-26-dark](https://github.com/user-attachments/assets/0ed3e895-d806-49bd-b142-6e7548757a83) |
| ![before-android-26-rose](https://github.com/user-attachments/assets/4ecf2500-a93a-46a3-bfce-181399d3a8ad) | ![after-android-26-rose](https://github.com/user-attachments/assets/47fd402e-e0b3-4828-83dc-646d94476652) |

Android 7, SDK 24

| Before | After |
| --- | --- |
| ![before-android-24-light](https://github.com/user-attachments/assets/d7f97fd3-3acd-4b1c-be67-f7c22df9ac00) | ![after-android-24-light](https://github.com/user-attachments/assets/3f2b3a4e-4d0c-4aba-adfa-b6b8e605f5d3) |
| ![before-android-24-dark](https://github.com/user-attachments/assets/f42c92d3-15eb-473a-8b5e-8dfa4205ed45) | ![after-android-24-dark](https://github.com/user-attachments/assets/1f19d8df-39a2-4c65-a6df-d294289d5262) |
| ![before-android-24-rose](https://github.com/user-attachments/assets/bff6c1bc-bbd6-4fb5-850c-19a0c06753c7) | ![after-android-24-rose](https://github.com/user-attachments/assets/9c8e3691-94bf-48f0-b252-ffc6aaa7f80b) |

## Checklist
 
#### I have tested any UI changes...
